addons/mail/security/mail_security.xml

```diff
@@ -25,9 +26,10 @@
             <field name="model_id" ref="model_mail_channel_partner"/>
             <field name="groups" eval="[(4, ref('base.group_user')), (4, ref('base.group_portal'))]"/>
             <field name="domain_force">['|', '|',
-('channel_id.public', '=', 'public'),
-'&amp;', ('channel_id.public', '=', 'private'), ('channel_id.is_member', '=', True),
-'&amp;', ('channel_id.public', '=', 'groups'), ('channel_id.group_public_id', 'in', [g.id for g in user.groups_id])]</field>
+                '&amp;', ('channel_id.channel_type', '!=', 'channel'), ('channel_id.is_member', '=', True),
+                '&amp;', ('channel_id.channel_type', '=', 'channel'), ('channel_id.group_public_id', '=', False),
+                '&amp;', ('channel_id.channel_type', '=', 'channel'), ('channel_id.group_public_id', 'in', [g.id for g in user.groups_id])]
+            </field>
```

Comment:
cleanup: instead of keeping the old format, just make one case for != channel, and one case for channel with the OR on group_public_id only inside it.